sdk/app_cpu1/common/drv/timing_manager.c

```diff
@@ -139,14 +146,39 @@ void timing_manager_send_manual_trigger(void)
 }
 
 /*
- * ISR for IRQ_F2P[0:0]. Called when sched_isr in timing
- * manager is set to 1, e.g. when all of the sensors are
- * done and the time has been collected.
+ * Sepcify the interrupt source of the scheduler ISR:
+ *
+ * Mode 0 uses the timing manager's 'trigger' signal, i.e. the control
+ * frequency based on the PWM carrier frequency and the specified user ratio.
+ *
+ * Mode 1 uses the timing manager's 'all_done' signal, calling the scheduler
+ * when all the sensors are done with acquisition. When no sensors are enabled,
+ * the scheduler is called as in mode 0 (based on the trigger). This mode
+ * supports reporting of the timing for each sensor
+ */
+void timing_manager_set_scheduler_source(void)
+{
+    uint32_t config_reg_addr = TIMING_MANAGER_BASE_ADDR + TIMING_MANAGER_ISR_REG_OFFSET;
+#if USER_CONFIG_ISR_SOURCE == 0
+    // Clear slv_reg4[1]
+    Xil_Out32(config_reg_addr, (Xil_In32(config_reg_addr) & 0xFFFFFFFD));
+#elif USER_CONFIG_ISR_SOURCE == 1
+    // Set slv_reg4[1]
+    Xil_Out32(config_reg_addr, (Xil_In32(config_reg_addr) | 0x00000002));
+#endif
+}
+
+/*
+ * ISR for IRQ_F2P[0:0]; updates the timing statistics for each
+ * sensor and updates the state of the scheduler
  */
-void isr_0(void *intc_inst_ptr)
+void timing_manager_isr(void *intc_inst_ptr)
 {
     // Push stats for each sensor
     timing_manager_sensor_stats();
+    // Increment elapsed time and run tasks
+    scheduler_tick();
+    // Clear the interrupt once done handling ISR
     timing_manager_clear_isr();
```

Comment:
Hmm, the fact that you only clear the ISR *after* you run all the tasks: does this ensure that, if there is a task overrun, the ISR won't actually fire? since it wasn't cleared?

Reply:
If the overrun error being ignored is an issue when testing the tasks, then we might need to make sure that all the tasks are run before the ISR is cleared. I think right now after setting the `scheduler_idle` state to false at the end of the function, it returns back to the ISR to clear the interrupt **then** goes back into `scheduler_run` to run all the tasks. So the tasks run after the ISR is cleared (not 100% sure though). If tasks are running, scheduler_idle is already false, and the only problem would be the elapsed time.

If this is a problem, and we want to keep the same structure (have the tasks running in the infinite loop in `scheduler_run`, I'm wondering if we could just use the condition for the time checking such that the `scheduler_tick` only "ticks" (updates elapsed time and scheduler state) when there's no tasks being overrun...or something like that

Reply:
Ah, you are right, my mistake. The `scheduler_tick()` function does not run anything in its context, just unblocks the task runner loop in the main context.

This looks like it should behave as desired